### PR TITLE
Transition demisto-sdk to the new docker registry proxy

### PR DESCRIPTION
**Pull request by samuelFain**
## Related Issues
relates: [CIAC-11589](https://jira-dc.paloaltonetworks.com/browse/CIAC-11589)

## Description
Transition demisto-sdk to a new docker registry proxy. details in the ticket.

### CODE REVIEW

File: demisto_sdk/commands/common/docker/dockerhub_client.py [hunk 1]
```diff
@@ -490,3 +513,118 @@ def get_repository_images_names(self, repo: str = DEFAULT_REPOSITORY) -> List[st
...
         ]
+
+
+### Google Artifactory functions ###
+
+
+def get_dockerhub_artifact_registry_url(base_path: str) -> str:
+    """
+    Parses a DockerHub Google Artifact Registry internal base path into a base url for DockerHub proxy API calls.
+    See https://confluence-dc.paloaltonetworks.com/display/VisibilityApplication/Shared-Services+GCP+Services+-+GAR for more details.
```
Comment (JasBeilin) [inline demisto_sdk/commands/common/docker/dockerhub_client.py:524]: See Confluence page: Shared-Services GCP Services - GAR

File: demisto_sdk/commands/common/docker_helper.py [hunk 2]
```diff
@@ -183,6 +263,7 @@ def __del__(self):
     @staticmethod
     @functools.lru_cache
     def version() -> Version:
+        logger.debug("version is called")
```
Comment (JasBeilin) [inline demisto_sdk/commands/common/docker_helper.py:266]: I think we can remove this.

**Review by JasBeilin — APPROVED** [review]
NICE! Please add changelog.

File: demisto_sdk/commands/common/docker_helper.py (hunk 2 above)
Comment (samuelFain) [reply demisto_sdk/commands/common/docker_helper.py]: Done.

File: demisto_sdk/commands/common/docker/dockerhub_client.py (hunk 1 above)
Comment (samuelFain) [reply demisto_sdk/commands/common/docker/dockerhub_client.py]: Done.

**Pull request merged**